tar/private/tar.bzl

```diff
@@ -474,29 +496,49 @@ def _mtree_impl(ctx):
             # copy workspace name here just in case to prevent ctx
             # to be transferred to execution phase.
             workspace_name = str(ctx.workspace_name)
+            format_each = "{}/%s".format(runfiles_dir)
 
-            content.add(_mtree_line(runfiles_dir, type = "dir"))
+            content.add(_mtree_line(runfiles_dir + "/", type = "dir"))
             content.add_all(
                 s.default_runfiles.empty_filenames,
-                format_each = "{}/%s".format(runfiles_dir),
+                format_each = format_each,
                 # be careful about what you pass to map_each as it will carry the data structures over to execution phase.
-                map_each = lambda f, e: _mtree_line(_vis_encode(f.removeprefix("external/") if f.startswith("external/") else workspace_name + "/" + f), "file"),
+                map_each = lambda f, e: _mtree_line(_vis_encode(f.removeprefix("../") if f.startswith("../") else workspace_name + "/" + f), "file"),
                 allow_closure = True,
             )
             content.add_all(
                 s.default_runfiles.files,
-                expand_directories = True,
                 uniquify = True,
-                format_each = "{}/%s".format(runfiles_dir),
+                format_each = format_each,
+                # be careful about what you pass to map_each as it will carry the data structures over to execution phase.
+                map_each = lambda f, e: _expand(f, e, _to_rlocation_path(f, workspace_name)),
+                allow_closure = True,
+            )
+            content.add_all(
+                s.default_runfiles.symlinks,
+                uniquify = True,
+                format_each = format_each,
                 # be careful about what you pass to map_each as it will carry the data structures over to execution phase.
-                map_each = lambda f, e: _expand(f, e, lambda f: _to_rlocation_path(f, workspace_name)),
+                map_each = lambda s, e: _expand(s.target_file, e, paths.normalize(workspace_name + "/" + s.path)),
```

Comment:
anything with `paths.` from skylib we had performance issues with big datasets (due to split and allocation of additional arrays) before so we rather not use it. is it necessary here?